src/test/java/edu/kis/powp/jobs2d/TestJobs2dPatterns.java

```diff
@@ -81,6 +84,18 @@ private static void setupLogger(Application application) {
 		application.addComponentMenuElement(Logger.class, "OFF logging", (ActionEvent e) -> logger.setLevel(Level.OFF));
 	}
 
+	private static void testJoe1Factory(){
+		Joe1Factory test = new Joe1Factory(new DrawerDroverAdapter());
+		test.setRectangle();
+		test.DrawRectangle();
+	}
+
+	private static void testJoe2Factory(){
+		Joe2Factory test = new Joe2Factory(new LineDrawerAdapter());
+		test.setRectangle();
+		test.DrawRectangle();
+	}
```

Comment:
Tests should be added using a convention in the original code (in User Tests menu )